tools/sloop.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * Library of functions pertaining to Sloop shadows. Slooping is a way
+ * to simulate the built-in snooping functionality of EotL.
+ *
+ * @author gon
+ */
+
+#define SLOOPEE_SHAD "/usr/gon/tools/sloopee_shad"
+#define SLOOPER_SHAD "/usr/gon/tools/slooper_shad"
+
+// Function Declarations
+protected void add_sloop_shadow( object ob, string str );
```

Comment:
There's still some confusion about the use of private and protected.
The functions you've marked here as `protected` I think you really want to be marked `private`. These functions are intended to be used by functions inside sloop.c, and only sloop.c. The functions you have specified as public (default) should be `protected`, as they are intended to be used by programs inheriting this one (such as sphere.c). They are not intended to be called by other objects via call_other().